mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -147,20 +217,46 @@ fn account_id_from_privkey(priv_key: &[u8], prefix: &str) -> MmResult<AccountId,
         .map_to_mm(|e| TendermintInitErrorKind::CouldNotGenerateAccountId(e.to_string()))
 }
 
-fn upper_hex(bytes: &[u8]) -> String {
+#[derive(Display, Debug)]
+pub enum AccountIdFromPubkeyHexErr {
+    InvalidHexString(FromHexError),
+    CouldNotCreateAccountId(ErrorReport),
+}
+
+impl From<FromHexError> for AccountIdFromPubkeyHexErr {
+    fn from(err: FromHexError) -> Self { AccountIdFromPubkeyHexErr::InvalidHexString(err) }
+}
+
+impl From<ErrorReport> for AccountIdFromPubkeyHexErr {
+    fn from(err: ErrorReport) -> Self { AccountIdFromPubkeyHexErr::CouldNotCreateAccountId(err) }
+}
+
+pub fn account_id_from_pubkey_hex(prefix: &str, pubkey: &str) -> MmResult<AccountId, AccountIdFromPubkeyHexErr> {
+    let pubkey_bytes = hex::decode(pubkey)?;
+    let pubkey_hash = dhash160(&pubkey_bytes);
+    Ok(AccountId::new(prefix, pubkey_hash.as_slice())?)
+}
+
+pub(crate) fn upper_hex(bytes: &[u8]) -> String {
     let mut str = hex::encode(bytes);
```

Comment:
If I'm not mistaken, there is `hex::encode_upper` function

Reply:
Good note, I added it, so let me do the fix 🙂

Reply:
Done